pkg/kubelet/cm/topologymanager/policy_options.go

```diff
@@ -71,11 +76,31 @@ func NewPolicyOptions(policyOptions map[string]string) (PolicyOptions, error) {
 				return opts, fmt.Errorf("bad value for option %q: %w", name, err)
 			}
 			opts.PreferClosestNUMA = optValue
+		case MaxAllowableNUMANodes:
+			optValue, err := strconv.Atoi(value)
+			if err != nil {
+				return opts, fmt.Errorf("bad value for option %q: %w", name, err)
+			}
+
+			if optValue < DefaultMaxAllowableNUMANodes {
+				return opts, fmt.Errorf("the minimum value of %q should not be less than %v", name, DefaultMaxAllowableNUMANodes)
+			}
+
+			if optValue > DefaultMaxAllowableNUMANodes {
+				klog.InfoS("WARNING: the value of max-allowable-numa-nodes is more than the default recommended value", "max-allowable-numa-nodes", optValue, "DefaultMaxAllowableNUMANodes", DefaultMaxAllowableNUMANodes)
+			}
+			opts.MaxAllowableNUMANodes = optValue
 		default:
 			// this should never be reached, we already detect unknown options,
 			// but we keep it as further safety.
 			return opts, fmt.Errorf("unsupported topologymanager option: %q (%s)", name, value)
 		}
 	}
+
+	// if user doesn't specify a new value, make defaultMaxAllowableNUMANodes
+	// is the recommended default.
+	if opts.MaxAllowableNUMANodes == 0 {
+		opts.MaxAllowableNUMANodes = DefaultMaxAllowableNUMANodes
+	}
```

Comment:
Can we instead move this to the top and do it unconditionally?, i.e.
```
// Set MaxAllowableNUMANodes to the default. This will be overwritten
// if the user has specified a policy option for MaxAllowableNUMANodes.
opts.MaxAllowableNUMANodes = DefaultMaxAllowableNUMANodes
```